x-pack/plugins/fleet/common/services/policy_template.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  RegistryPolicyTemplate,
+  RegistryPolicyInputOnlyTemplate,
+  RegistryPolicyIntegrationTemplate,
+  RegistryInput,
+  PackageInfo,
+  RegistryVarsEntry,
+  RegistryDataStream,
+} from '../types';
+
+const DATA_STREAM_DATASET_VAR: RegistryVarsEntry = {
```

Comment:
We will always want to make dataset customisable for input-only packages, but the packages do not define it as a variable because they are not concerned with streams only input.

this means we have to add the variable in ourselves